lang/attribute/event/src/lib.rs

```diff
@@ -81,6 +81,35 @@ pub fn emit(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     })
 }
 
+#[proc_macro]
+pub fn emit_cpi(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let tuple = parse_macro_input!(input as syn::ExprTuple);
+
+    let elems = tuple.elems;
+    if elems.len() != 2 {
+        panic!("Expected a tuple with exactly two elements.");
+    }
+
+    let first = &elems[0];
+    let second = &elems[1];
+
+    proc_macro::TokenStream::from(quote! {
+        let program_info: &anchor_lang::solana_program::account_info::AccountInfo = &#first;
+
+        let __disc: Vec<u8> = crate::event::EVENT_IX_TAG_LE.to_vec();
+        let __inner_data: &Vec<u8> = &anchor_lang::Event::data(&#second);
```

Comment:
Should be able to remove the extra references here:
```rs
let __inner_data: Vec<u8> = anchor_lang::Event::data(#second);
```

Reply:
Fixed